### ISSUE TITLE
Question: will it save disk storage in scorch with DocValues disabled?

### ISSUE THREAD

**Issue by darren** [blevesearch/bleve#1279]
We want to reduce disk usage by scorch,  So I wonder whether disable DocValues will do, but experimenting with this shows that DocValues has no impact on disk usage:

Are there any ways to reduce scorch disk usage?

This is my test:
```go
package main

import (
	"fmt"
	"log"
	"strconv"
	"time"

	"github.com/blevesearch/bleve"
	"github.com/blevesearch/bleve/index/scorch"
)

type Message struct {
	Content string `json:"content"`
}

func (m *Message) Type() string {
	return "message"
}

func createIndex(name string, enableDocValues bool) bleve.Index {
	mapping := bleve.NewIndexMapping()

	messageMapping := bleve.NewDocumentMapping()
	mapping.AddDocumentMapping("message", messageMapping)

	contentMapping := bleve.NewTextFieldMapping()
	contentMapping.DocValues = enableDocValues

	messageMapping.AddFieldMappingsAt("content", contentMapping)

	idx, err := bleve.NewUsing(name, mapping, scorch.Name, scorch.Name, nil)
	if err != nil {
		panic(err)
	}

	return idx
}

func main() {
	idx := createIndex("doc.bleve", true)
	nidx := createIndex("ndoc.bleve", false)

	defer idx.Close()
	defer nidx.Close()

	var b *bleve.Batch
	var nb *bleve.Batch

	for i := 0; i < 1000000; i++ {
		id := strconv.Itoa(i)
		c := &Message{fmt.Sprintf("%d %v", i, time.Now())}

		if i%1000 == 0 {
			if b != nil {
				idx.Batch(b)
			}
			if nb != nil {
				nidx.Batch(nb)
			}

			log.Println(i)
			b = idx.NewBatch()
			nb = nidx.NewBatch()
		}
		b.Index(id, c)
		nb.Index(id, c)
	}

	log.Println("Waiting...")
	time.Sleep(1 * time.Minute)
}

```
Disk Usage:

```
$ ls -hlaS  doc.bleve/store/
total 3303784
-rw-------   1 darren  wheel   167M Aug 16 12:20 000000000136.zap
-rw-------   1 darren  wheel   164M Aug 16 12:21 0000000002d8.zap
-rw-------   1 darren  wheel   164M Aug 16 12:20 000000000199.zap
-rw-------   1 darren  wheel   164M Aug 16 12:20 000000000207.zap
-rw-------   1 darren  wheel   161M Aug 16 12:21 00000000026a.zap
-rw-------   1 darren  wheel   153M Aug 16 12:20 0000000000b1.zap
-rw-------   1 darren  wheel   153M Aug 16 12:21 0000000003a5.zap
-rw-------   1 darren  wheel   140M Aug 16 12:21 00000000033b.zap
-rw-------   1 darren  wheel   135M Aug 16 12:21 000000000406.zap
-rw-------   1 darren  wheel    61M Aug 16 12:20 000000000043.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 000000000438.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 00000000042e.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 000000000442.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 00000000041b.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 000000000424.zap
-rw-------   1 darren  wheel   6.7M Aug 16 12:21 00000000041d.zap
-rw-------   1 darren  wheel   6.0M Aug 16 12:21 00000000044c.zap
-rw-------   1 darren  wheel   5.2M Aug 16 12:21 000000000455.zap
-rw-------   1 darren  wheel   625K Aug 16 12:21 00000000045a.zap
-rw-------   1 darren  wheel   624K Aug 16 12:21 000000000459.zap
-rw-------   1 darren  wheel   624K Aug 16 12:21 000000000458.zap
-rw-------   1 darren  wheel   623K Aug 16 12:21 00000000045b.zap
-rw-------   1 darren  wheel   622K Aug 16 12:21 000000000456.zap
-rw-------   1 darren  wheel   621K Aug 16 12:21 000000000457.zap
-rw-------   1 darren  wheel   128K Aug 16 12:21 root.bolt
drwx------  27 darren  wheel   864B Aug 16 12:21 .
drwx------   4 darren  wheel   128B Aug 16 12:19 ..

$ ls -hlaS  ndoc.bleve/store/
total 3265064
-rw-------   1 darren  wheel   165M Aug 16 12:20 000000000136.zap
-rw-------   1 darren  wheel   162M Aug 16 12:20 000000000207.zap
-rw-------   1 darren  wheel   162M Aug 16 12:20 000000000199.zap
-rw-------   1 darren  wheel   162M Aug 16 12:21 0000000002d8.zap
-rw-------   1 darren  wheel   159M Aug 16 12:21 00000000026a.zap
-rw-------   1 darren  wheel   151M Aug 16 12:21 0000000003a5.zap
-rw-------   1 darren  wheel   151M Aug 16 12:20 0000000000b1.zap
-rw-------   1 darren  wheel   138M Aug 16 12:21 00000000033b.zap
-rw-------   1 darren  wheel   133M Aug 16 12:21 000000000406.zap
-rw-------   1 darren  wheel    59M Aug 16 12:20 000000000043.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 000000000438.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 00000000042e.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 000000000442.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 00000000041a.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 000000000424.zap
-rw-------   1 darren  wheel   6.5M Aug 16 12:21 00000000041c.zap
-rw-------   1 darren  wheel   5.7M Aug 16 12:21 00000000044c.zap
-rw-------   1 darren  wheel   5.0M Aug 16 12:21 000000000455.zap
-rw-------   1 darren  wheel   601K Aug 16 12:21 000000000459.zap
-rw-------   1 darren  wheel   601K Aug 16 12:21 00000000045a.zap
-rw-------   1 darren  wheel   601K Aug 16 12:21 000000000458.zap
-rw-------   1 darren  wheel   600K Aug 16 12:21 00000000045b.zap
-rw-------   1 darren  wheel   599K Aug 16 12:21 000000000456.zap
-rw-------   1 darren  wheel   598K Aug 16 12:21 000000000457.zap
-rw-------   1 darren  wheel   128K Aug 16 12:21 root.bolt
drwx------  27 darren  wheel   864B Aug 16 12:21 .
drwx------   4 darren  wheel   128B Aug 16 12:19 ..
```

Zap Info:

```
bleve zap docvalue doc.bleve/store/000000000136.zap 
FieldID: 0 '_id' docvalue at 18446744073709551615 (ffffffffffffffff) not  persisted 
FieldID: 1 '_all' docvalue at 18446744073709551615 (ffffffffffffffff) not  persisted 
FieldID: 2 'content' docvalue at 172653254 (a4a7ac6) numChunks 98  diskSize 120.926053 MB
Total Doc Values Size on Disk: 120.926053 MB

bleve  zap docvalue ndoc.bleve/store/000000000136.zap 
FieldID: 0 '_id' docvalue at 18446744073709551615 (ffffffffffffffff) not  persisted 
FieldID: 1 '_all' docvalue at 18446744073709551615 (ffffffffffffffff) not  persisted 
FieldID: 2 'content' docvalue at 18446744073709551615 (ffffffffffffffff) not  persisted 
Total Doc Values Size on Disk: 0.000000 MB
```

**Comment by hubyhuby** [comment]
hi @darren ,
Never used "contentMapping.DocValues = enableDocValues"
But I suppose what you want to do is : contentMapping.Store = false
It s what I understand from the mapping doc here : https://blevesearch.com/docs/Index-Mapping/

**Comment by mschoch** [comment]
I ran the code you provided (with bleve master) and I do observed that turning off doc values saves some space.

```
du -sh doc.bleve ndoc.bleve
1.6G	doc.bleve
1.5G	ndoc.bleve
```

As @hubyhuby said, turning off stored values can also save space.

We are aware that scorch has some significant storage inefficiencies, and you can see some of the work on that here: https://github.com/blevesearch/bleve/pull/1277  I ran your example using that branch (by accident because I forgot I was on it) and observed these sizes:

```
du -sh doc.bleve ndoc.bleve 
579M	doc.bleve
552M	ndoc.bleve
```
I do not recommend using this branch in production, as we will not be merging this to master until we have a better story around versioning the bleve index format.  But it gives some idea the direction we're going.

**Comment by darren** [comment]
I tested #1277  on our documents  (3000,000 docs) , 
with scorch master it takes 61G disk size after index,
while #1277 only takes 8.0GB disk size,  about 8 times in disk savings!

**Comment by mschoch** [comment]
@darren we are eager to get that merged to master, but we have to work out the versioning first, as we need backwards compat with the current scorch/zap file format.

We are working on it, as you have seen the space savings are significant.

**Comment by mschoch** [comment]
The new segment version 12 which makes these space savings possible is now available with Bleve 1.0.x.  It is not enabled by default, you need to create your index with the configuration:

```
"forceSegmentType": "zap",
"forceSegmentVersion": 12
```
